src/sentry/migrations/0281_add_new_indexer_table.py

```diff
@@ -0,0 +1,60 @@
+# Generated by Django 2.2.24 on 2022-03-16 18:33
+
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not. Generally
+    # we don't want to run in a transaction here, since for long running operations like data
+    # back-fills this results in us locking an increasing number of rows until we finally commit.
+    atomic = False
+
+    dependencies = [
+        ("sentry", "0280_extend_commit_author_email_length"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="StringIndexer",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("string", models.CharField(max_length=200)),
+                ("organization_id", sentry.db.models.fields.bounded.BoundedBigIntegerField()),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                (
+                    "last_seen",
+                    models.DateTimeField(db_index=True, default=django.utils.timezone.now),
+                ),
+            ],
+            options={
+                "db_table": "sentry_stringindexer",
+            },
+        ),
+        migrations.AddConstraint(
+            model_name="stringindexer",
+            constraint=models.UniqueConstraint(
+                fields=("string", "organization_id"), name="unique_org_string"
+            ),
+        ),
+    ]
```

Comment:
Please add a `retention_days` column. We will not have to enforce the same retention for all index keys. For example we will have to delete after 30 days for GDPR